da/celestia/celestia.go

```diff
@@ -111,60 +127,101 @@ func (c *DataAvailabilityLayerClient) SubmitBlock(ctx context.Context, block *ty
 }
 
 // CheckBlockAvailability queries DA layer to check data availability of block at given height.
-func (c *DataAvailabilityLayerClient) CheckBlockAvailability(
-	ctx context.Context, dataLayerHeight uint64,
-) da.ResultCheckBlock {
-	shares, err := c.client.NamespacedShares(ctx, c.namespaceID, dataLayerHeight)
-	code := dataRequestErrorToStatus(err)
-	if code != da.StatusSuccess {
+func (c *DataAvailabilityLayerClient) CheckBlockAvailability(ctx context.Context, dataLayerHeight uint64) da.ResultCheckBlock {
+	header, err := c.rpc.Header.GetByHeight(ctx, dataLayerHeight)
+	if err != nil {
+		return da.ResultCheckBlock{
+			BaseResult: da.BaseResult{
+				Code:    da.StatusError,
+				Message: err.Error(),
+			},
+		}
+	}
+	if header.DAH == nil {
+		return da.ResultCheckBlock{
+			BaseResult: da.BaseResult{
+				Code:     da.StatusSuccess,
+				DAHeight: dataLayerHeight,
+			},
+			DataAvailable: false,
+		}
+	}
+	err = c.rpc.Share.SharesAvailable(ctx, header.DAH)
+	if err != nil {
+		if err.Error() == ErrNotAvailable.Error() {
+			return da.ResultCheckBlock{
+				BaseResult: da.BaseResult{
+					Code:     da.StatusSuccess,
+					DAHeight: dataLayerHeight,
+				},
+				DataAvailable: false,
+			}
+		}
 		return da.ResultCheckBlock{
 			BaseResult: da.BaseResult{
 				Code:    da.StatusError,
 				Message: err.Error(),
 			},
 		}
 	}
-
 	return da.ResultCheckBlock{
 		BaseResult: da.BaseResult{
 			Code:     da.StatusSuccess,
 			DAHeight: dataLayerHeight,
 		},
-		DataAvailable: len(shares) > 0,
+		DataAvailable: true,
 	}
```

Comment:
This could be simplified if you turn my other comment into a helper

```go 
func isSpecificError(err, specificError error) bool {
	return errors.Is(err, specificError) || strings.Contains(err.Error(), specificError.Error())
}
```

Then you can do this
```go	
        notAvailable := isSpecificError(err, ErrNotAvailable)
	if err != nil && !notAvailable {
		return da.ResultCheckBlock{
			BaseResult: da.BaseResult{
				Code:    da.StatusError,
				Message: err.Error(),
			},
		}
	}

	return da.ResultCheckBlock{
		BaseResult: da.BaseResult{
			Code:     da.StatusSuccess,
			DAHeight: dataLayerHeight,
		},
		DataAvailable: !notAvailable,
	}
```

Reply:
Using `strings.Contains` for now as the error type can't be imported without a dependency.